### PR TITLE
[quant][graphmode][fx][fix] Fix support for custom module

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59042 [quant][graphmode][fx][refactor] Remove _convert from Quantizer class
* **#59041 [quant][graphmode][fx][fix] Fix support for custom module**
* #59040 [quant][graphmode][fx][refactor] Remove Quantizer class from convert (QuantizeHandler)
* #59039 [quant][graphmode][fx][refactor] Remove Quantizer class from quantize_node
* #59038 [quant][graphmode][fx][refactor] Remove _prepare from Quantizer class
* #59037 [quant][graphmode][fx][refactor] Remove find_matches from Quantizer class
* #59036 [quant][graphmode][fx][refactor] Remove fold_weights from Quantizer class
* #59035 [quant][graphmode][fx][refactor] Remove save_state and restore_state from Quantizer class
* #59034 [quant][graphmode][fx][refactor] Remove prepare_custom_config from Quantizer class
* #59033 [quant][graphmode][fx][refactor] Remove patterns from Quantizer class
* #59032 [quant][graphmode][fx][refactor] Remove node_name_to_scope from Quantizer
* #59031 [quant][graphmode][fx][refactor] Remove quantized_graph from Quantizer
* #59030 [quant][gx][graphmode][refactor] Remove modules from Quantizer
* #59029 [quant][graphmode][fx] Factor out run_weight_observer
* #59028 [quant][graphmode][fx] Merge quant_env and env

Summary:
Static quantization for Custom module support was removed in a previous refactor
https://github.com/pytorch/pytorch/pull/57519 since it's not covered by the test case
This PR re-enabled the test case and fixed the support

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724866](https://our.internmc.facebook.com/intern/diff/D28724866)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724866).

File: torch/quantization/fx/quantization_patterns.py [hunk 1]
```diff
@@ -1305,13 +1305,11 @@ def convert(self,
...
         assert custom_module_class_mapping is not None
         observed_custom_module = modules[str(node.target)]
-        # TODO: remove quantizer here
         if activation_is_statically_quantized(qconfig):
-            assert node.name in quantizer.activation_post_process_map  # type: ignore[name-defined]
-            cur_idx = quantizer.activation_post_process_indexes[node.name]  # type: ignore[name-defined]
-            observed_custom_module.activation_post_process = \
-                modules[quantizer.activation_post_process_map[node.name][cur_idx]]  # type: ignore[name-defined]
-            quantizer.activation_post_process_indexes[node.name] += 1  # type: ignore[name-defined]
+            activation_post_process = \
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:1309]: thanks, sorry for missing this in my PR

**Review by vkuzo — APPROVED** [review]

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724866).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724866).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724866).

**Pull request closed without merging**